File: src/main/java/com/rsupport/stringtable/StringTableGenerator.java
package com.rsupport.stringtable;

import org.apache.poi.xssf.usermodel.XSSFSheet;
import org.apache.poi.xssf.usermodel.XSSFWorkbook;

import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;

public class StringTableGenerator {
    public static void generate(File source, String pathRes, String targetSheetName, int indexRowNum) throws IOException {
        System.out.println("Generate string tables.");
        System.out.println("\tsource: " + source.getPath());
        System.out.println("\tres: " + pathRes);
        File res = new File(pathRes);
        FileInputStream inputStream = new FileInputStream(source);
        XSSFWorkbook workbook = new XSSFWorkbook(inputStream);

        Sheet2Strings.getInstance().convert(getTargetSheet(targetSheetName, workbook), res, indexRowNum);
        System.out.println("Completed.");
        inputStream.close();
    }

    private static XSSFSheet getTargetSheet(String targetSheetName, XSSFWorkbook workbook) {
        if (targetSheetName.isEmpty()) {
            return workbook.getSheetAt(0);
        }

        XSSFSheet resultSheet = workbook.getSheet(targetSheetName);
        assert (resultSheet != null) : "Not Found target sheet: " + targetSheetName;
        return resultSheet;
    }
}

File: src/main/java/com/rsupport/stringtable/StringTableGenerator.kt
package com.rsupport.stringtable

import org.apache.poi.xssf.usermodel.XSSFSheet
import org.apache.poi.xssf.usermodel.XSSFWorkbook
import java.io.File
import java.io.FileInputStream

object StringTableGenerator {

    @Throws(Exception::class)
    fun generate(
        source: String,
        resPath: String,
        targetSheetName: String,
        rowPositionColumnHeader: Int = 1,
        xmlFileName: String = "strings_generated.xml",
        doNotConvertNewLine: Boolean = false,
        defaultLanguageForValues: String = "en",
    ) {
        println("Generate string tables.")
        println("\tsource: $source")
        println("\tres: $resPath")
        val sourceFile = File(source)
        val pathRes = File(resPath)
        val inputStream = FileInputStream(sourceFile)
        val workbook = XSSFWorkbook(inputStream)
        Sheet2Strings.convert(
            sheet = getTargetSheet(targetSheetName, workbook),
            pathRes = pathRes,
            rowPositionColumnHeader = rowPositionColumnHeader,
            defaultLanguageForValues = defaultLanguageForValues,
            outputXmlFileName = xmlFileName,
            doNotConvertNewLine = doNotConvertNewLine,
        )
        println("Completed.")
        inputStream.close()
    }

    private fun getTargetSheet(targetSheetName: String?, workbook: XSSFWorkbook): XSSFSheet {
        if (targetSheetName.isNullOrEmpty())
            return workbook.getSheetAt(0)

        return workbook.getSheet(targetSheetName).also { result ->
            assert(result != null) { "Not found target sheet: $targetSheetName" }
        }
    }
}